src/net_processing.cpp

```diff
@@ -2685,8 +2685,10 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
 
         // Potentially mark this peer as a preferred download peer.
         {
-        LOCK(cs_main);
-        UpdatePreferredDownload(pfrom, State(pfrom.GetId()));
+            LOCK(cs_main);
+            CNodeState* state = State(pfrom.GetId());
```

Comment:
I think the commit _[net processing] Inline and simplify UpdatePreferredDownload_ could have slightly more explanation in the commit log (explaining that this logic can only be hit once per peer, so `fPreferredDownload` will always be `false`, and the `m_num_preferred_download_peers -= state->fPreferredDownload;` line is always a no-op).